### PR TITLE
Issue #765

### PR DESCRIPTION
**Pull request by KadekM**
In first commit I removed Guard which would never throw.

In second commit I removed guards and changed them to manual exception throwing.

Should I also remove Guard class since it's now no longer in use ?

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
looks good to me - how about you @HCanber ?

File: src/core/Akka.Remote/DeadlineFailureDetector.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public DeadlineFailureDetector(TimeSpan acceptableHeartbeatPause, Clock clock =
         {
             _acceptableHeartbeatPause = acceptableHeartbeatPause;
             _acceptableHeartbeatMillis = Convert.ToInt64(acceptableHeartbeatPause.TotalMilliseconds);
-            Guard.Assert(_acceptableHeartbeatPause > TimeSpan.Zero, "acceptable-heartbeat-pause must be greater than zero");
+            if (_acceptableHeartbeatPause <= TimeSpan.Zero) throw new ArgumentException("acceptable-heartbeat-pause must be greater than zero");
```
Comment (rogeralsing) [inline src/core/Akka.Remote/DeadlineFailureDetector.cs]: If the value refers to a setting in a config, the type should be `Akka.Configuration.ConfigurationException`
The hyphen'ed name in the message hints that this is the case.

**Comment by rogeralsing** [comment]
Looks good, IMO we should probably supply parameter names to the Argument*Exceptions, not just bake them into the exception message, as we do provide param names in the rest of the codebase.

File: src/core/Akka.Remote/DeadlineFailureDetector.cs (hunk 1 above)
Comment (Aaronontheweb) [inline src/core/Akka.Remote/DeadlineFailureDetector.cs]: Ah yes, that's correct - should be an `Akka.Configuration.ConfigurationException`. Could you please change that @KadekM ?

**Comment by KadekM** [comment]
I will fix it ASAP (today / tommorow) . So far thank you for review. I'm kind of new to open source.

**Comment by Aaronontheweb** [comment]
thanks @KadekM! No worries - most of the people working on this project were new to open source before they started sending PRs ;) you're in good company.

**Comment by KadekM** [comment]
Great :+1:
So should I also delete Akka.Util.Guard since it's now no longer in use ?

**Comment by KadekM** [comment]
Test is passing on my machine - also I don't suppose change in my commit would cause this fail, is the test somewhat unstable ?

**Comment by KadekM** [comment]
Ok, I ammended my commit (added second parameters to Argument*Exception) and now two different tests are failing... what's going on ? :)

**Comment by Aaronontheweb** [comment]
@KadekM racey unit tests - not your fault. I'll re-run the build.

**Comment by KadekM** [comment]
Ok. Is there anything else need to be done?

**Comment by Aaronontheweb** [comment]
@KadekM nope, good to go! thanks for your pr!

**Pull request merged**